### PR TITLE
Use partial semantic model in import completion

### PR DESCRIPTION
**Pull request by genlu**
Fixes [AB#1275634](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1275634)

### CODE REVIEW

File: src/Features/CSharp/Portable/Completion/CompletionProviders/ImportCompletion/ImportCompletionProviderHelper.cs [hunk 1]
```diff
@@ -21,10 +22,12 @@ internal static class ImportCompletionProviderHelper
             => semanticModel.GetUsingNamespacesInScope(location)
                 .SelectAsArray(namespaceSymbol => namespaceSymbol.ToDisplayString(SymbolDisplayFormats.NameFormat));
 
-        public static async Task<SyntaxContext> CreateContextAsync(Document document, int position, CancellationToken cancellationToken)
+        public static async Task<SyntaxContext?> CreateContextAsync(Document document, int position, CancellationToken cancellationToken)
         {
             // Need regular semantic model because we will use it to get imported namespace symbols. Otherwise we will try to 
             // reach outside of the span and ended up with "node not within syntax tree" error from the speculative model.
+            var sourceText = await document.GetTextAsync(cancellationToken).ConfigureAwait(false);
+            document = sourceText.GetDocumentWithFrozenPartialSemantics(cancellationToken) ?? document;
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/CompletionProviders/ImportCompletion/ImportCompletionProviderHelper.cs:30]: 1. should comment what we're doing this for.
2. `?? doc` should be unnecessary.  we should throw if this fails.

File: src/Features/VisualBasic/Portable/Completion/CompletionProviders/ImportCompletionProvider/ImportCompletionProviderHelper.vb [hunk 2]
```diff
@@ -33,6 +34,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Completion.Providers
         Public Shared Async Function CreateContextAsync(document As Document, position As Integer, cancellationToken As CancellationToken) As Task(Of SyntaxContext)
             ' Need regular semantic model because we will use it to get imported namespace symbols. Otherwise we will try to 
             ' reach outside of the span And ended up with "node not within syntax tree" error from the speculative model.
+            Dim sourceText = Await document.GetTextAsync(cancellationToken).ConfigureAwait(False)
+            Dim frozenDocument = sourceText.GetDocumentWithFrozenPartialSemantics(cancellationToken)
+            If frozenDocument Is Nothing Then
+                document = frozenDocument
+            End If
```
Comment (CyrusNajmabadi) [inline src/Features/VisualBasic/Portable/Completion/CompletionProviders/ImportCompletionProvider/ImportCompletionProviderHelper.vb:41]: if you wanted `??` equivalent, it's `if(x, y)` in VB.  but we def don't want this in either event :)

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs [hunk 3]
```diff
@@ -116,7 +116,8 @@ private async Task CheckTokenAsync(CodeFixContext context, Document document, Sy
             var originalOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
             var options = originalOptions
                 .WithChangedOption(CompletionOptions.SnippetsBehavior, document.Project.Language, SnippetsRule.NeverInclude)
-                .WithChangedOption(CompletionOptions.ShowItemsFromUnimportedNamespaces, document.Project.Language, false);
+                .WithChangedOption(CompletionOptions.ShowItemsFromUnimportedNamespaces, document.Project.Language, false)
+                .WithChangedOption(CompletionServiceOptions.IsExpandedCompletion, false);
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs:120]: doc.

**Comment by CyrusNajmabadi** [comment]
> Use partial semantic model in import completion

can we move to this for *all* completion providers?

**Comment by genlu** [comment]
> can we move to this for all completion providers?

I can do this in a follow up PR.

File: src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs (hunk 3 above)
Comment (genlu) [reply src/Features/Core/Portable/SpellCheck/AbstractSpellCheckCodeFixProvider.cs:120]: Will add in a follow up PR

**Pull request merged**